RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc

```diff
@@ -111,2143 +114,2222 @@ using namespace edm;
-	    theCurrentTemp.head.fbin[0] = fbin0;
-	    theCurrentTemp.head.fbin[1] = fbin1;
-	    theCurrentTemp.head.fbin[2] = fbin2;
-	  }
-		
-	  LOGINFO("SiPixelTemplate") << "Template ID = " << theCurrentTemp.head.ID << ", Template Version " << theCurrentTemp.head.templ_version << ", Bfield = " << theCurrentTemp.head.Bfield 
-				     << ", NTy = " << theCurrentTemp.head.NTy << ", NTyx = " << theCurrentTemp.head.NTyx<< ", NTxx = " << theCurrentTemp.head.NTxx << ", Dtype = " << theCurrentTemp.head.Dtype
-		<< ", Bias voltage " << theCurrentTemp.head.Vbias << ", temperature "
-		<< theCurrentTemp.head.temperature << ", fluence " << theCurrentTemp.head.fluence << ", Q-scaling factor " << theCurrentTemp.head.qscale
-		<< ", 1/2 multi dcol threshold " << theCurrentTemp.head.s50 << ", 1/2 single dcol threshold " << theCurrentTemp.head.ss50
+   
+   //  open the template file
+   
+   std::ifstream in_file(tempfile, std::ios::in);
+   
+   if(in_file.is_open()) {
+      
+      // Create a local template storage entry
+      
+      SiPixelTemplateStore theCurrentTemp;
+      
+      // Read-in a header string first and print it
+      
+      for (i=0; (c=in_file.get()) != '\n'; ++i) {
+         if(i < 79) {theCurrentTemp.head.title[i] = c;}
+      }
+      if(i > 78) {i=78;}
+      theCurrentTemp.head.title[i+1] ='\0';
+      LOGINFO("SiPixelTemplate") << "Loading Pixel Template File - " << theCurrentTemp.head.title << ENDL;
+      
+      // next, the header information
+      
+      in_file >> theCurrentTemp.head.ID  >> theCurrentTemp.head.templ_version >> theCurrentTemp.head.Bfield >> theCurrentTemp.head.NTy >> theCurrentTemp.head.NTyx >> theCurrentTemp.head.NTxx
+      >> theCurrentTemp.head.Dtype >> theCurrentTemp.head.Vbias >> theCurrentTemp.head.temperature >> theCurrentTemp.head.fluence >> theCurrentTemp.head.qscale
+      >> theCurrentTemp.head.s50 >> theCurrentTemp.head.lorywidth >> theCurrentTemp.head.lorxwidth >> theCurrentTemp.head.ysize >> theCurrentTemp.head.xsize >> theCurrentTemp.head.zsize;
+      
+      if(in_file.fail()) {LOGERROR("SiPixelTemplate") << "Error reading file 0A, no template load" << ENDL; return false;}
+      
+      if(theCurrentTemp.head.templ_version > 17) {
+         in_file >> theCurrentTemp.head.ss50 >> theCurrentTemp.head.lorybias
+         >> theCurrentTemp.head.lorxbias >> theCurrentTemp.head.fbin[0]
+         >> theCurrentTemp.head.fbin[1] >> theCurrentTemp.head.fbin[2];
+         
+         if(in_file.fail()) {LOGERROR("SiPixelTemplate") << "Error reading file 0B, no template load"
+            << ENDL; return false;}
+      } else {
+         theCurrentTemp.head.ss50 = theCurrentTemp.head.s50;
+         theCurrentTemp.head.lorybias = theCurrentTemp.head.lorywidth/2.f;
+         theCurrentTemp.head.lorxbias = theCurrentTemp.head.lorxwidth/2.f;
+         theCurrentTemp.head.fbin[0] = 1.5f;
+         theCurrentTemp.head.fbin[1] = 1.00f;
+         theCurrentTemp.head.fbin[2] = 0.85f;
```

Comment:
just to clarify on my comment, your change here, once you get past all the reindentation, is 
```diff
-	   theCurrentTemp.head.fbin[0] = fbin0;
-	   theCurrentTemp.head.fbin[1] = fbin1;
-	   theCurrentTemp.head.fbin[2] = fbin2;
+         theCurrentTemp.head.fbin[0] = 1.5f;
+         theCurrentTemp.head.fbin[1] = 1.00f;
+         theCurrentTemp.head.fbin[2] = 0.85f;
```
So, I'm just asking to go back to what was here before since it's a more appropriate style for magic numbers.